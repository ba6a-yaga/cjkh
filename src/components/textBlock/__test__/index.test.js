import React from 'react';
import ReactDOM from 'react-dom';
import TextBlock from '../index';
import renderer from "react-test-renderer"
it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TextBlock title={"title"} desc={"description"}></TextBlock>, div);
    ReactDOM.unmountComponentAtNode(div);
});


it("render correctly with title", () => {
    const TextInputComponent = renderer.create(<TextBlock title={"Заголовок"}></TextBlock>).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
});

it("render correctly with title and description", () => {
    const TextInputComponent = renderer.create(<TextBlock title={"Заголовок"} desc={"Описание"}></TextBlock>).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
});

it("render correctly with title, description and button", () => {
    const TextInputComponent = renderer.create(<TextBlock title={"Заголовок"} desc={"Текст описания"} button={"КНОПКА"}></TextBlock>).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
});
