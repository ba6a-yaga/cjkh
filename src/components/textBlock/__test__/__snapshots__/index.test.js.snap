// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly with title 1`] = `
<div
  className="by-text-block "
>
  <h1
    className="by-title-text"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Заголовок",
      }
    }
  />
  <div
    className="by-description-text"
  />
</div>
`;

exports[`render correctly with title and description 1`] = `
<div
  className="by-text-block "
>
  <h1
    className="by-title-text"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Заголовок",
      }
    }
  />
  <div
    className="by-description-text"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Описание",
      }
    }
  />
</div>
`;

exports[`render correctly with title, description and button 1`] = `
<div
  className="by-text-block "
>
  <h1
    className="by-title-text"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Заголовок",
      }
    }
  />
  <div
    className="by-description-text"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Текст описания",
      }
    }
  />
  <button
    className="by-btn  "
    onClick={[Function]}
  >
    <p>
      КНОПКА
    </p>
    
  </button>
</div>
`;
